shared/metrics/src/main/java/io/pravega/shared/metrics/DynamicLoggerImpl.java

```diff
@@ -95,6 +95,24 @@ public Counter call() throws Exception {
     }
 
     @Override
+    public long getCounterValue(String name) {
+        Exceptions.checkNotNullOrEmpty(name, "name");
+        String counterName = name + ".Counter";
+        try {
+            Counter counter = countersCache.get(counterName, new Callable<Counter>() {
+                @Override
+                public Counter call() throws Exception {
+                    return underlying.createCounter(counterName);
+                }
+            });
+            return counter.get();
```

Comment:
That's right because this metrics code doesn't have access to anything else, so if it is not in its local cache, then it will be reset. I don't think there is a way around it, unless we add some callback mechanism to enable the metrics code to request an uncached value. What do you guys think?